packages/tools/client-debugger/client-debugger-view/src/components/TelemetryView.tsx

```diff
@@ -64,6 +69,37 @@ export function TelemetryView(): React.ReactElement {
 		ITimestampedTelemetryEvent[] | undefined
 	>();
 	const [maxEventsToDisplay, setMaxEventsToDisplay] = React.useState<number>(DEFAULT_PAGE_SIZE);
+	const [selectedCategory, setSelectedCategory] = useState("");
+	const [filteredTelemetryEvents, setFilteredTelemetryEvents] = React.useState<
+		ITimestampedTelemetryEvent[] | undefined
+	>();
+	const [customSearch, setCustomSearch] = React.useState("");
+	const [eventNameOptions, setEventNameOptions] = useState<string[]>([]);
+	const [matchingOptions, setMatchingOptions] = React.useState([...eventNameOptions]);
+
+	React.useEffect(() => {
+		// Create list of all event names
+		setEventNameOptions([
+			...new Set(
+				telemetryEvents?.map((event) =>
+					event.logContent.eventName.slice("fluid:telemetry:".length),
+				),
+			),
+		]);
+		// Initialy matching options are all options
+		setMatchingOptions(eventNameOptions.sort());
+		const filteredEvents = telemetryEvents?.filter((event) => {
+			return (
+				(selectedCategory === "" ||
+					selectedCategory === "All" ||
+					event.logContent.category === selectedCategory) &&
+				(customSearch === "" ||
+					event.logContent.eventName.slice("fluid:telemetry:".length) === customSearch)
```

Comment:
Personal preference, but I think it's worth splitting this from one large chained set boolean operations into some if else statements. It'll be much easier to understand and maintain it in the future.